src/components/Company/CompanyName/CorrectNameRequest.vue

```diff
@@ -100,32 +141,35 @@ export default class CorrectNameRequest extends Mixins(NameRequestMixin) {
   }
 
   private resetForm () {
-    this.nameRequestNumber = ''
-    this.entityPhone = ''
-    this.entityEmail = ''
     this.$refs.correctNrForm.resetValidation()
   }
 
   /** Watch for form submission and emit results. */
-  @Watch('submit')
+  @Watch('submitId')
   private async onSubmit (): Promise<any> {
-    await this.validateNameRequest(this.nameRequestNumber, this.entityPhone, this.entityEmail)
-      .then(response => {
-        const nrCorrection: NrCorrectionIF = {
-          nrNumber: this.nameRequestNumber,
-          legalName: this.getNrApprovedName(response)
-        }
-        this.setNameRequest(nrCorrection)
-        this.emitDone(true)
-      }).catch(() => {
-        this.emitDone()
-      })
+    if (this.submitId === CorrectionTypes.CORRECT_NEW_NR) {
+      await this.validateNameRequest(this.nameRequestNumber, this.entityPhone, this.entityEmail)
+        .then(response => {
+          const nrCorrection: NrCorrectionIF = {
+            nrNumber: this.nameRequestNumber,
+            legalName: this.getNrApprovedName(response),
+            applicant: {
+              phoneNumber: this.entityPhone,
+              emailAddress: this.entityEmail
+            }
+          }
+          this.setNameRequest({ ...this.nameRequest, ...nrCorrection })
+          this.emitDone(CorrectionTypes.CORRECT_NEW_NR)
+        }).catch(() => {
+          this.emitDone()
```

Comment:
No error feedback to user?
No console log for debugging?

Reply:
the error is handled during the request. If it fails during the request the dialog will appear with the type of error. This is simply to inform the parent that process ( whether success or fail ) is now complete.

Reply:
Oh right. And I think I asked you about this previously.
Could you please add a comment saying the request handles its own errors?

Reply:
Of course!